contracts/lending/NFTFiLending.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "./NFTLendingBase.sol";
+import "./interfaces/nftfi/IDirectLoanFixedCollectionOffer.sol";
+import "./interfaces/nftfi/IDirectLoanCoordinator.sol";
+
+/// @title NFTfi Lending
+/// @notice Manages creating and repaying a loan on NFTfi
+contract NFTfiLending is NFTLendingBase {
+    using SafeERC20 for IERC20;
+
+    /// @notice NFTfi DirectLoanFixedCollectionOffer Contract
+    IDirectLoanFixedCollectionOffer
+        public immutable directLoanFixedCollectionOffer;
+
+    /// @notice NFTfi DirectLoanCoordinator Contract
+    IDirectLoanCoordinator public immutable directLoanCoordinator;
+
+    /// @notice NFTFILending Constructor
+    /// @param _directLoanFixedCollectionOffer DirectLoanFixedCollectionOffer contract address
+    constructor(
+        address _globalBNPL,
+        IDirectLoanFixedCollectionOffer _directLoanFixedCollectionOffer,
+        IDirectLoanCoordinator _directLoanCoordinator
+    ) NFTLendingBase(_globalBNPL) {
+        directLoanFixedCollectionOffer = _directLoanFixedCollectionOffer;
+        directLoanCoordinator = _directLoanCoordinator;
+    }
+
+    /// @inheritdoc INFTLending
+    function borrow(
+        bytes calldata _inputData
+    ) external onlyBNPL returns (uint256) {
+        // Decode `inputData` into Offer, Signature and BorrowerSettings
+        (
+            IDirectLoanFixedCollectionOffer.Offer memory offer,
+            IDirectLoanFixedCollectionOffer.Signature memory signature,
+            IDirectLoanFixedCollectionOffer.BorrowerSettings
+                memory borrowerSettings
+        ) = abi.decode(
+                _inputData,
+                (
+                    IDirectLoanFixedCollectionOffer.Offer,
+                    IDirectLoanFixedCollectionOffer.Signature,
+                    IDirectLoanFixedCollectionOffer.BorrowerSettings
+                )
+            );
+
+        IERC721 nft = IERC721(offer.nftCollateralContract);
+        uint256 nftId = offer.nftCollateralId;
+
+        // Transfer NFT from BNPL contract
+        nft.safeTransferFrom(msg.sender, address(this), nftId);
+
```

Comment:
Im thinking if this is gonna cost us even more gas. Like right now basically we will buy the NFT from the BNPL contract, transfer here and start a loan.